[RFC] Fix compiler warnings by initialising variables.

amateurhourrrr opened the pull request:
Two variables were warned about potentially being unused at compilation time. This PR initialises the variables, even though there were no code paths in which the variables are left unset.

DarkDeepBlue committed:
vim-patch:8.0.0074

Problem:    Cannot make Vim fail on an internal error.
Solution:   Add IEMSG() and IEMSG2(). (Domenique Pelle)  Avoid reporting an
            internal error without mentioning where.

https://github.com/vim/vim/commit/95f096030ed1a8afea028f2ea295d6f6a70f466f

Signed-off-by: Michael Schupikov <michael@schupikov.de>

hiecaq committed:
health.vim: Try `pyenv root` #7341

bfredl committed:
provider: delete vimL stderr collector, now that it exists builtin

bfredl committed:
channel: check for existance before trying to set key

This avoids an error message in async context, where it is not safe.

janlazo committed:
ci: nodejs client acceptance-test #7706

ci: install nodejs 8 in Appveyor, Travis

provider: check node version for debug support
Resolve https://github.com/neovim/neovim/pull/7577#issuecomment-350590592 for Unix.

provider: test if nodejs in ci supports --inspect-brk

nodejs host for neovim requires nodejs 6+ to work properly.
nodejs 6.12+ or 7.6+ is required for debug support via `node --inspect-brk`.

provider: run cli.js of nodejs host directly

npm shims are useless because the user cannot set node to debug mode via
--inspect-brk. This is problematic on Windows which use batchfiles and
shell scripts to compensate for not supporting shebang.

The patch uses `npm root -g` to get the absolute path of the global npm
modules. If that fails, then the user did not install neovim npm package
globally. Use that absolute path to find `neovim/bin/cli.js`, which is
what the npm shim actually runs with node. glob() is for a simple file
check in case bin/ is removed because the npm shims are ignored now.

ZyX-I committed:
eval/typval: Add functions useful for hiding list implementation

ZyX-I committed:
eval/typval: Add macros useful for hiding list item implementation

ZyX-I committed:
*: Start hiding list implementation

Most of files, except for eval.c and eval/* were only processed by perl.

ZyX-I committed:
eval: Fix inputlist()

ZyX-I committed:
eval: Fix writefile()

ZyX-I committed:
*: Hide list implementation in other files as well

ZyX-I committed:
eval,functests: Fix tests and complete() and setline() behaviour

ZyX-I committed:
eval: Fix setmatches(), setqflist() and setloclist()

ZyX-I committed:
quickfix: Fix :cexpr and :lexpr

ZyX-I committed:
functests: Mark islocked("v:_null_list") behaviour correct

It is the same for other VAR_FIXED lists.

ZyX-I committed:
eval/encode: Fix crash in json_encode test suite

ZyX-I committed:
eval: Fix uniq() crash in legacy test 055

ZyX-I committed:
functests: Add some more NULL tests

ZyX-I committed:
eval: Still check for NULL when doing :unlet

ZyX-I committed:
*: Fix test failures

ZyX-I committed:
*: Finish hiding list implementation

ZyX-I committed:
*: Fix linter errors

ZyX-I committed:
eval: Fix add()

ZyX-I committed:
eval/typval: Fix typo

[ci skip]

ZyX-I committed:
Merge branch 'master' into hide-container-impl

ZyX-I committed:
*: Fix linter errors

ZyX-I committed:
clint,eval: Make linter check for direct usage of list attributes

ZyX-I committed:
*: Fix some problems found during review

Still missing: problems in window.c, it should be possible to construct a test
for them.

amateurhourrrr committed:
eval.c: Fix compiler warning by initialising variable.

There are no code paths for which the variable is not set, however the
compiler was unaware of this.

amateurhourrrr committed:
eval.c: Fix compiler warning by initialising variable.

There are no code paths for which the variable is not set, however the
compiler was unaware of this.

amateurhourrrr committed:
window.c: Fix compiler warning by initialising variable.

There were no code paths for which this variable is not set, however the
compiler was unaware of this.

amateurhourrrr committed:
window.c: Fix compiler warning by initialising variable.

There were no code paths for which this variable is not set, however the
compiler was unaware of this.

ZyX-I committed:
window: Fix matchaddpos() and enhance error reporting

ZyX-I committed:
eval/encode: Avoid unnecessary tv_list_idx_of_item() calls

justinmk commented:
Maybe the optimizer removed a codepath, resulting in the variables not being initialized.

gebulmer commented:
It doesn't look like it based on the disassembly, for both cases, they both reach the inner scopes which set them.

But as I'm not an assembly wizard someone else should check this also.

jamessan committed:
lint

justinmk committed:
tui: rework deferred-termcodes ... again

- Revert timer-based approach.
- Instead, call loop_poll_events() with a timeout in an "active" loop,
  to infer that "TUI startup activity has mostly finished", but also to
  enforce a mininum time (100 ms) before emitting "enable focus
  reporting" termcode. (If TUI startup takes longer than that minimum
  time, it's probably a slow environment anyways.)
- Tickle `main_loop` by sending a dummy event.  Without this, the
  initial "focus-gained" response from the terminal may not get
  processed until the user hits a key.

ref #7720
ref #7664
ref #7649
ref #7664
ref 27f9b1c7b029d8

jamessan committed:
Merge pull request #7306 from DarkDeepBlue/vim-8.0.0074

vim-patch:8.0.0074

ZyX-I committed:
eval: Fix linter error

ZyX-I committed:
Merge branch 'master' into hide-container-impl

ZyX-I committed:
functests: Fix linter error

janlazo committed:
test: remove inspect test; set NODE_PATH in nodejs_spec.lua

provider#node#can_inspect will fail on some systems because it is common
to have old node versions in OS (any Linux OS that has LTS releases)
and CI (Travis, Appveyor).

NODE_PATH can be trivially set with VimL.
Build scripts don't have to set it for the nodejs tests to work.
NODE_PATH is optional to begin with and is used only as a workaround
for the neovim node.js host.

jamessan committed:
vim-patch.sh: Include upstream summary in commit message

[ci skip]

jamessan committed:
vim-patch:8.0.0420: text garbled when the system encoding differs from 'encoding'

Problem:    When running :make the output may be in the system encoding,
            different from 'encoding'.
Solution:   Add the 'makeencoding' option. (Ken Takata)

https://github.com/vim/vim/commit/2c7292dc5bbf155fe2192d417363b8c085759cad

ZyX-I committed:
eval: Fix some issues found in review

justinmk committed:
health.vim: nodejs: skip if nodejs is too old

justinmk committed:
provider/nodejs: check version in Detect()

justinmk committed:
Merge #7706 'ci: nodejs acceptance-test'

jamessan committed:
lint

alexgenco committed:
health.vim: mention g:ruby_host_prog #7737

jamessan committed:
Merge pull request #7732 from jamessan/patch-summary

vim-patch.sh: Include upstream summary in commit message

justinmk committed:
ASAN/LeakSanitizer: ignore loop_schedule_deferred()

clang ASAN/LeakSanitizer error (observed in #7706):

    ==21832==ERROR: LeakSanitizer: detected memory leaks
    Direct leak of 56 byte(s) in 1 object(s) allocated from:
        0 0x511b26 in malloc (/home/travis/build/neovim/neovim/build/bin/nvim+0x511b26)
        1 0x1009a84 in try_malloc /home/travis/build/neovim/neovim/src/nvim/memory.c:87:15
        2 0x1009c44 in xmalloc /home/travis/build/neovim/neovim/src/nvim/memory.c:121:15
        3 0xaa8c36 in loop_schedule_deferred /home/travis/build/neovim/neovim/src/nvim/event/loop.c:89:19
        4 0x190856a in tui_main /home/travis/build/neovim/neovim/src/nvim/tui/tui.c:367:5
        5 0x1963d61 in ui_thread_run /home/travis/build/neovim/neovim/src/nvim/ui_bridge.c:106:3
        6 0x2b5d4190d183 in start_thread /build/eglibc-SvCtMH/eglibc-2.19/nptl/pthread_create.c:312

Possible explanation: During exit, `Loop.thread_events` may not get
flushed, so `loop_deferred_event()` is never called.

We could instead try to unwind `Loop.thread_events` during teardown, but
it seems lower-risk to just tell ASAN to ignore it.

Valgrind does not complain:
    $ while :; do { 2>valglog.txt valgrind ./build/bin/nvim -u NONE +q ; } ; if ! [ $? = 0 ] ; then break ; fi ; done

justinmk committed:
Merge #7739 'ASAN/LeakSanitizer: ignore loop_schedule_deferred()'

justinmk committed:
provider/nodejs: more robust version-check (#7738)

jamessan committed:
Merge pull request #7736 from jamessan/vim-8.0.0420

[RFC] vim-patch:8.0.0420: text garbled when the system encoding differs from 'encoding'

jamessan committed:
vim-patch:8.0.0404: not enough testing for quickfix

Problem:    Not enough testing for quickfix.
Solution:   Add some more tests. (Yegappan Lakshmanan)

https://github.com/vim/vim/commit/391b1dd040af204b150d43c5a1c97477ee450a28

jamessan committed:
vim-patch:8.0.0484: :lhelpgrep does not fail after a successful one

Problem:    Using :lhelpgrep with an argument that should fail does not
            produce an error if the previous :helpgrep worked.
Solution:   Use another way to detect that autocommands made the quickfix info
            invalid. (Yegappan Lakshmanan)

https://github.com/vim/vim/commit/ee85df37634dfb0c40ae5de0b4f246aef460b392

jamessan committed:
Merge pull request #7740 from jamessan/vim-8.0.0404

[RFC] vim-patch:8.0.0404,8.0.0484

jamessan committed:
vim-patch:8.0.0517: there is no way to remove quickfix lists

Problem:    There is no way to remove quickfix lists (for testing).
Solution:   Add the 'f' action to setqflist(). Add tests. (Yegappan
            Lakshmanan)

https://github.com/vim/vim/commit/b6fa30ccc39cdb7f1d07b99fe2f4c6b61671dac2

jamessan committed:
vim-patch:8.0.0536: quickfix window not updated when freeing quickfix stack

Problem:    Quickfix window not updated when freeing quickfix stack.
Solution:   Update the quickfix window. (Yegappan Lakshmanan)

https://github.com/vim/vim/commit/69f40be64555d50f603c6f22722cf762aaa6bbc1

jamessan committed:
vim-patch:8.0.0584: memory leak when executing quickfix tests

Problem:    Memory leak when executing quickfix tests.
Solution:   Free the list reference. (Yegappan Lakshmanan)

https://github.com/vim/vim/commit/d788f6fe89c77262c474de323f5dab6d1c814e27

concatime committed:
Updating to latest UNIBILIUM (#7745)

Update to unibilium 1.2.1

jamessan committed:
Merge pull request #7744 from jamessan/vim-8.0.0517

[RFC] vim-patch:8.0.0517,8.0.0536,8.0.0584

jamessan committed:
vim-patch:8.0.0565: using freed memory in :caddbuf

Problem:    Using freed memory in :caddbuf after clearing quickfix list.
            (Dominique Pelle)
Solution:   Set qf_last to NULL.

https://github.com/vim/vim/commit/31bdd13c335533c749993b57dcd980a87373139e

jamessan committed:
vim-patch:8.0.0574: get only one quickfix list after :caddbuf

Problem:    Get only one quickfix list after :caddbuf.
Solution:   Reset qf_multiline. (Yegappan Lakshmanan)

https://github.com/vim/vim/commit/99895eac1cf71be43ece7e14b50e206e041fbe9f

jamessan committed:
vim-patch:8.0.0579: duplicate test case for quickfix

Problem:    Duplicate test case for quickfix.
Solution:   Remove the function. (Yegappan Lakshmanan)

https://github.com/vim/vim/commit/9b77016545d5ef1a1f4a90c9bb4b7a6693af8918

jamessan committed:
vim-patch:8.0.0580: cannot set the valid flag with setqflist()

Problem:    Cannot set the valid flag with setqflist().
Solution:   Add the "valid" argument. (Yegappan Lakshmanan, closes vim/vim#1642)

https://github.com/vim/vim/commit/f1d21c8cc83f40c815b6bf13cd2043152db533ee

jamessan committed:
Merge pull request #7747 from jamessan/vim-8.0.0565

vim-patch:8.0.0565,8.0.0574,8.0.0579,8.0.0580

jamessan committed:
vim-patch:8.0.0590: cannot add a context to locations

Problem:    Cannot add a context to locations.
Solution:   Add the "context" entry in location entries. (Yegappan Lakshmanan,
            closes vim/vim#1012)

https://github.com/vim/vim/commit/8f77c5a4ec756f3f866bd6b18feb6fca6f2a2e91

jamessan committed:
vim-patch:8.0.0595: Coverity warning for not checking return value

Problem:    Coverity warning for not checking return value of dict_add().
Solution:   Check the return value for FAIL.

https://github.com/vim/vim/commit/beb9cb19c660484488a71a25eda46ab0fa579278

jamessan committed:
vim-patch:8.0.0597: off-by-one error in size computation

Problem:    Off-by-one error in buffer size computation.
Solution:   Use ">=" instead of ">". (Lemonboy, closes vim/vim#1694)

https://github.com/vim/vim/commit/253f9128779f315ea670f9b4a17446b7b4c74927

jamessan committed:
vim-patch:8.0.0606: cannot set the context for a specified quickfix list

Problem:    Cannot set the context for a specified quickfix list.
Solution:   Use the list index instead of the current list. (Yegappan
            Lakshmanan)

https://github.com/vim/vim/commit/6e62da3e14d32f76f60d5cc8b267059923842f17

jamessan committed:
Merge pull request #7751 from jamessan/vim-8.0.0590

[RFC] vim-patch:8.0.0590,8.0.0595,8.0.0597,8.0.0606

ZyX-I committed:
Merge branch 'master' into hide-container-impl

ZyX-I committed:
regexp: Fix linter errors

ZyX-I committed:
clint: Make linter report line where it found opening brace

justinmk committed:
Merge #7646 from bfredl/chan_buffered

Document and defer error message when buffered stream would overwrite channels dict key

justinmk committed:
Merge #7708 from ZyX-I/hide-container-impl

amateurhourrrr committed:
Merge branch 'compilerwarnings' of https://github.com/gebulmer/neovim into compilerwarnings

gebulmer commented:
Closing as it's trivial enough for someone to include this in their ongoing work in eval and window.
